File: chars.ts
declare global {
  interface Window {
    __TAURI_INVOKE__<T>(
      cmd: string,
      args?: Record<string, unknown>
    ): Promise<T>;
  }
}
const invoke = window.__TAURI_INVOKE__;
/**
 * A function that accepts a character
 */
export async function takeChar(x: string): Promise<void> {
  await invoke<void>("plugin:chars|0", { 0: x });
}
/**
 * A function that returns a character
 */
export async function returnChar(): Promise<string> {
  const result = await invoke<string>("plugin:chars|1");
  return result;
}

File: chars.rs
#[allow(clippy::all, unused)]
pub mod imports {
    /// A function that accepts a character
    pub async fn take_char(x: char) -> () {
        #[derive(::serde::Serialize)]
        #[serde(rename_all = "camelCase")]
        struct Params {
            #[serde(rename = "0")]
            x: char,
        }
        let params = Params { x };
        ::tauri_bindgen_guest_rust::invoke("plugin:chars|0", &params).await
    }
    /// A function that returns a character
    pub async fn return_char() -> char {
        ::tauri_bindgen_guest_rust::invoke("plugin:chars|1", ()).await
    }
}
